### PR TITLE
ELY-2776 Add tests to the OIDC testsuite to test for cases where request/request_uri parameters are not supported by the OpenID provider

### PR DESCRIPTION
**Pull request by PrarthonaPaul**
Issue: https://issues.redhat.com/browse/ELY-2776

### CODE REVIEW

File: http/oidc/src/test/java/org/wildfly/security/http/oidc/MockOidcClientConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
+package org.wildfly.security.http.oidc;
```
Comment (fjuma) [inline http/oidc/src/test/java/org/wildfly/security/http/oidc/MockOidcClientConfiguration.java:19]: Missing copyright header.

File: http/oidc/src/test/java/org/wildfly/security/http/oidc/MockOidcClientConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
+import static org.wildfly.security.http.oidc.KeycloakConfiguration.ALICE;
+import static org.wildfly.security.http.oidc.KeycloakConfiguration.ALICE_PASSWORD;
+import static org.wildfly.security.http.oidc.Oidc.AuthenticationRequestFormat.REQUEST;
+import static org.wildfly.security.http.oidc.Oidc.AuthenticationRequestFormat.REQUEST_URI;
+import static org.wildfly.security.http.oidc.Oidc.OIDC_NAME;
+import static org.wildfly.security.http.oidc.Oidc.OIDC_SCOPE;
+
+@RunWith(JMockit.class)
```
Comment (fjuma) [inline http/oidc/src/test/java/org/wildfly/security/http/oidc/MockOidcClientConfiguration.java:61]: Would be good to have some documentation about what the test class is doing, how it's making use of the mocked config, etc.

File: http/oidc/src/test/java/org/wildfly/security/http/oidc/MockOidcClientConfiguration.java (hunk 1 above)
Comment (PrarthonaPaul) [reply http/oidc/src/test/java/org/wildfly/security/http/oidc/MockOidcClientConfiguration.java:19]: fixed!

File: http/oidc/src/test/java/org/wildfly/security/http/oidc/MockOidcClientConfiguration.java (hunk 2 above)
Comment (PrarthonaPaul) [reply http/oidc/src/test/java/org/wildfly/security/http/oidc/MockOidcClientConfiguration.java:61]: fixed!

**Review by fjuma — APPROVED** [review]
Thanks @PrarthonaPaul!

**Pull request merged**